pkg/oauthserver/oauthserver/oauth_apiserver.go

```diff
@@ -112,18 +108,35 @@ func getSessionSecrets(filename string) ([]string, error) {
 		}
 
 		for _, s := range sessionSecrets.Secrets {
-			secrets = append(secrets, s.Authentication)
-			secrets = append(secrets, s.Encryption)
+			// TODO make these length independent
+			secrets = append(secrets, []byte(s.Authentication))
+			secrets = append(secrets, []byte(s.Encryption))
 		}
 	} else {
 		// Generate random signing and encryption secrets if none are specified in config
-		secrets = append(secrets, fmt.Sprintf("%x", md5.Sum([]byte(uuid.NewRandom().String()))))
-		secrets = append(secrets, fmt.Sprintf("%x", md5.Sum([]byte(uuid.NewRandom().String()))))
+		secrets = append(secrets, randomBytes(64))
+		secrets = append(secrets, randomBytes(32))
 	}
 
 	return secrets, nil
 }
 
+func randomBytes(size int) []byte {
+	b := make([]byte, size)
+	if _, err := rand.Read(b); err != nil {
+		panic(err) // rand should never fail
+	}
+	return b
+}
+
+func randomString(size int) string {
+	// each byte (8 bits) gives us 4/3 base64 (6 bits) characters
+	// we account for that conversion and add one to handle truncation
+	b64size := base64.RawURLEncoding.DecodedLen(size) + 1
```

Comment:
Raw bytes to string via base64 makes the resulting string longer.  It is not always possible to pick a base64 size that will expand to the extract string size requested.  This handles that by requesting a longer byte array than is required and then truncating to the requested string size (otherwise sometimes you would get a string that is one character shorter than requested).

Reply:
The q. is:
why do we care about the exact length of a base64 string, instead of the entropy we want ?
If base64 is a problem, let's use something that doesn't have the random expansion problem, like hexlify or something.